hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSetProcessor.java

```diff
@@ -729,7 +739,18 @@ private static Column createFilterCondition(Dataset dataset,String[] allColIdInS
                 continue; // Cannot Push Down Qualifier
             Column col = dataset.col(allColIdInSpark[q.getStoragePosition()]);
             q.clearOrderableCache();
-            Object value = (probeValue==null || i!=0?q.getOrderable():probeValue).getObject();
+
+            DataValueDescriptor dvd = probeValue;
+            if (dvd == null || i != 0) {
+                dvd = q.getOrderable();
+            }
+
+            if (dvd instanceof  SQLChar) {
+                col = adaptSQLCharColumn(col, (SQLChar) dvd);
+            }
+
+//            Object value = (probeValue==null || i!=0?q.getOrderable():probeValue).getObject();
```

Comment:
Please remove the commented out lines

Reply:
I made the change, and answered the question on baseColumMap in Mark's thread. Thanks!